src/util/segmentation/operations/correction.js

```diff
@@ -0,0 +1,68 @@
+import { fillInside } from '.';
+
+/*
+With the correction tool you draw a stroke and the tool does "something useful"
+http://mitk.org/wiki/Interactive_segmentation
+- Stroke starts and ends inside a segmentation -> something is added
+- Stroke starts and ends outside a segmentation -> something is removed
+- In and out several times -> above points are done for individual segments
+
+
+You do not have to draw a closed contour to use the Correction tool and do not need to switch between the Add and Subtract tool to perform small corrective changes. The following figure shows the usage of this tool:
+
+- if the user draws a line which starts and ends outside the segmentation AND it intersects no other segmentation the endpoints of the line are connected and the resulting contour is filled
+- if the user draws a line which starts and ends outside the segmentation a part of it is cut off (left image)
+- if the line is drawn fully inside the segmentation the marked region is added to the segmentation (right image)
+- http://docs.mitk.org/2016.11/org_mitk_views_segmentation.html
+ */
+
+export default function correction(
+  points,
+  segmentationData,
+  image,
+  labelValue = 1
+) {
+  const { width } = image;
+
+  // For each point, determine whether or not it is inside a segment
+  points.forEach(point => {
+    const { x, y } = point;
+    const xRound = Math.round(x);
+    const yRound = Math.round(y);
+
+    point.segment = segmentationData[yRound * width + xRound];
```

Comment:
Once we will come back to correction Tool soon, may I leave it for later?

Reply:
vNext is a bit of a playground currently, so that's fine. So long as the method doesn't have hidden side effects when it hits master, because then people start to depend on unintended functionality (e.g. piggy backing on this call just to get the label values for a set of points, even though there are other ways to do this).